### PR TITLE
Fix IndexError on strange backslash docstring

### LINKED ISSUE #505: IndexError on one-line backslash docstring
```python
def f():
    """a\
"""
```

```console
$ pydocstyle --version
5.1.0
$ pydocstyle t.py 
Traceback (most recent call last):
  File "/tmp/x/venv/bin/pydocstyle", line 8, in <module>
    sys.exit(main())
  File "/tmp/x/venv/lib/python3.8/site-packages/pydocstyle/cli.py", line 68, in main
    sys.exit(run_pydocstyle())
  File "/tmp/x/venv/lib/python3.8/site-packages/pydocstyle/cli.py", line 44, in run_pydocstyle
    errors.extend(check((filename,), select=checked_codes,
  File "/tmp/x/venv/lib/python3.8/site-packages/pydocstyle/checker.py", line 979, in check
    for error in ConventionChecker().check_source(source, filename,
  File "/tmp/x/venv/lib/python3.8/site-packages/pydocstyle/checker.py", line 123, in check_source
    for error in errors:
  File "/tmp/x/venv/lib/python3.8/site-packages/pydocstyle/checker.py", line 288, in check_indent
    indents[-1] > indent):
IndexError: list index out of range
```

### PR DESCRIPTION
**Pull request by asottile**
Thanks for submitting a PR!

Please make sure to check for the following items:
- [x] Add unit tests and integration tests where applicable.  
      If you've added an error code or changed an error code behavior,
      you should probably add or change a test case file under `tests/test_cases/` and add 
      it to the list under `tests/test_definitions.py`.  
      If you've added or changed a command line option,
      you should probably add or change a test in `tests/test_integration.py`.
- [x] Add a line to the release notes (docs/release_notes.rst) under "Current Development Version".  
      Make sure to include the PR number after you open and get one.
   
Please don't get discouraged as it may take a while to get a review.

Resolves #505

### CODE REVIEW

File: src/pydocstyle/checker.py [hunk 1]
```diff
@@ -284,10 +284,12 @@ def check_indent(self, definition, docstring):
                 indents = [leading_space(l) for l in lines if not is_blank(l)]
                 if set(' \t') == set(''.join(indents) + indent):
                     yield violations.D206()
-                if (len(indents) > 1 and min(indents[:-1]) > indent or
-                        indents[-1] > indent):
+                if (
+                    len(indents) > 1 and min(indents[:-1]) > indent or
```
Comment (samj1912) [inline src/pydocstyle/checker.py:288]: Probably worth putting brackets for clarity here?

File: src/pydocstyle/checker.py [hunk 2]
```diff
@@ -284,10 +284,12 @@ def check_indent(self, definition, docstring):
...
+                    len(indents) > 0 and indents[-1] > indent
+                ):
                     yield violations.D208()
-                if min(indents) < indent:
+                if len(indents) > 0 and min(indents) < indent:
```
Comment (samj1912) [inline src/pydocstyle/checker.py:292]: replace len(indents) > 0

with if indents and....

**Review by samj1912 — CHANGES_REQUESTED** [review]

File: src/pydocstyle/checker.py (hunk 2 above)
Comment (asottile) [reply src/pydocstyle/checker.py:292]: So, I wanted this to match the `> 1` above for clarity, it looks weirder not doing that -- I can change it if you insist but I felt it better to be consistent

File: src/pydocstyle/checker.py (hunk 1 above)
Comment (asottile) [reply src/pydocstyle/checker.py:288]: there weren't brackets before, the newline I think is enough to differentiate, I can change it if you insist (or you can push to the PR if you like)

**Review by ssbarnea — APPROVED** [review]

**Comment by ssbarnea** [comment]
@samj1912 Can you please accept the patch and create new tag?

**Review by samj1912 — APPROVED** [review]

**Comment by samj1912** [comment]
Will release the new version tonight :)

**Review by samj1912 — APPROVED** [review]

**Comment by samj1912** [comment]
Released :)

**Pull request merged**